src/plugins/discover/public/application/doc/components/doc.tsx

```diff
@@ -25,11 +27,31 @@ export interface DocProps extends EsDocSearchProps {
 
 export function Doc(props: DocProps) {
   const { dataView } = props;
-  const [reqState, hit] = useEsDocSearch(props);
   const services = useDiscoverServices();
   const { locator, chrome, docLinks } = services;
   const indexExistsLink = docLinks.links.apis.indexExists;
 
+  const onBeforeFetch = useCallback(() => {
+    return services.profilesManager.resolveDataSourceProfile({
+      dataSource: dataView?.id ? createDataViewDataSource({ dataViewId: dataView.id }) : undefined,
+      dataView,
+      query: { query: '', language: 'kuery' },
+    });
+  }, [services.profilesManager, dataView]);
+
+  const onProcessRecord = useCallback(
+    (record) => {
+      return services.profilesManager.resolveDocumentProfile({ record });
+    },
+    [services.profilesManager]
+  );
```

Comment:
@davismcphee QQ: what about the root context? Should we add it for these pages or not? I guess, for Dashboard we didn't resolve the root context.

Reply:
Sure, thanks for the link!